resources/aws/peering.go

```diff
@@ -0,0 +1,127 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/giantswarm/microerror"
+)
+
+const (
+	RequesterVpcFilterName = "requester-vpc-info.vpc-id"
+	AccepterVpcFilterName  = "accepter-vpc-info.vpc-id"
+)
+
+type VPCPeeringConnection struct {
+	VPCId     string // VPCId is the ID of the VPC in the guest cluster.
+	PeerVPCId string // PeerVPCId the ID of the VPC in the host cluster.
+	id        string
+	AWSEntity
+}
+
+func (v VPCPeeringConnection) findExisting() (*ec2.VpcPeeringConnection, error) {
+	vpcPeeringConnections, err := v.Clients.EC2.DescribeVpcPeeringConnections(
+		&ec2.DescribeVpcPeeringConnectionsInput{
+			Filters: []*ec2.Filter{
+				{
+					Name: aws.String(RequesterVpcFilterName),
+					Values: []*string{
+						aws.String(v.VPCId),
+					},
+				},
+				{
+					Name: aws.String(AccepterVpcFilterName),
+					Values: []*string{
+						aws.String(v.PeerVPCId),
+					},
+				},
+			},
+		},
+	)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	if len(vpcPeeringConnections.VpcPeeringConnections) < 1 {
+		return nil, microerror.Maskf(notFoundError, notFoundErrorFormat, VPCPeeringConnectionType, v.id)
+	} else if len(vpcPeeringConnections.VpcPeeringConnections) > 1 {
+		return nil, microerror.Mask(tooManyResultsError)
+	}
+
+	return vpcPeeringConnections.VpcPeeringConnections[0], nil
+}
+
+func (v *VPCPeeringConnection) checkIfExists() (bool, error) {
+	_, err := v.findExisting()
+	if IsNotFound(err) {
+		return false, nil
+	} else if err != nil {
+		return false, microerror.Mask(err)
+	}
+
+	return true, nil
+}
+
+func (v *VPCPeeringConnection) CreateIfNotExists() (bool, error) {
+	exists, err := v.checkIfExists()
+	if err != nil {
+		return false, microerror.Mask(err)
+	}
+
+	if exists {
+		return false, nil
+	}
+
+	if err := v.CreateOrFail(); err != nil {
+		return false, microerror.Mask(err)
+	}
+
+	return true, nil
+}
+
+func (v *VPCPeeringConnection) CreateOrFail() error {
+	vpcPeeringConnection, err := v.Clients.EC2.CreateVpcPeeringConnection(
+		&ec2.CreateVpcPeeringConnectionInput{
+			VpcId:     &v.VPCId,
+			PeerVpcId: &v.PeerVPCId,
+		},
+	)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	v.id = *vpcPeeringConnection.VpcPeeringConnection.VpcPeeringConnectionId
+
+	// TODO: Accept VPC peering connection from host cluster.
```

Comment:
When?

Reply:
This will be the next PR. We need the host cluster clients for this, so I'm doing it in a separate PR to keep the scope down.

Reply:
Can the connection be created before the peering request has been accepted? Or will it error?